sample/src/main/kotlin/io/getstream/chat/sample/feature/channels/ChannelsFragment.kt

```diff
@@ -37,7 +37,11 @@ class ChannelsFragment : Fragment(R.layout.fragment_channels) {
 
     private fun setupOnClickListeners() {
         channelsListView.setOnChannelClickListener {
-            findNavController().navigate(ChannelsFragmentDirections.actionOpenChannel(it.cid))
+            with(findNavController()) {
+                if (currentDestination?.id == R.id.channelsFragment) {
+                    navigate(ChannelsFragmentDirections.actionOpenChannel(it.cid))
```

Comment:
Don't we need this for every navigation we ever trigger? Maybe we should do something like [these extensions](https://github.com/woocommerce/woocommerce-android/pull/2401/files#diff-e012971d96c7f53a38c50392606490de)?